collectors/CMPCollector.js

```diff
@@ -258,6 +314,47 @@ class CMPCollector extends BaseCollector {
         }
     }
 
+    /**
+     * @returns {Promise<ScanResult>}
+     */
+    async scanPatterns() {
+        /**
+         * @type {string[]}
+         */
+        const foundPatterns = [];
+        const foundSnippets = [];
+        const pages = await this.context.pages();
+        if (pages.length > 0) {
+            const page = pages[0];
+            /**
+             * @type {Promise<string>[]}
+             */
+            const promises = [];
+            page.frames().forEach(frame => {
+                // eslint-disable-next-line no-undef
+                promises.push(frame.evaluate(() => document.documentElement.innerText).catch(reason => {
+                    this.log(`error retrieving text: ${reason}`);
+                    // ignore exceptions
+                    return '';
+                }));
+            });
+            const texts = await Promise.all(promises);
+            const allTexts = texts.join('\n');
+            for (const p of DETECT_PATTERNS) {
+                const matches = allTexts.match(p);
+                if (matches) {
+                    foundPatterns.push(p.toString());
+                    foundSnippets.push(...matches);
```

Comment:
let's put some limit on those matches because on nationalgeographic.com we are getting this:
![nationalgeographic](https://user-images.githubusercontent.com/985504/192587548-998d03a9-b965-4cdd-8fd0-259251f2dc8e.jpg)

90% of the output file looks like that :)

Reply:
oh jeez, I thought I checked this... thanks for catching!